bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/handler/HomeConnectOvenHandler.java

```diff
@@ -0,0 +1,323 @@
+
+            if (STATE_POWER_ON.equals(event.getValue())) {
+                updateChannels();
+            } else {
+                resetProgramStateChannels();
+                getThingChannel(CHANNEL_SELECTED_PROGRAM_STATE)
+                        .ifPresent(c -> updateState(c.getUID(), UnDefType.UNDEF));
+                getThingChannel(CHANNEL_ACTIVE_PROGRAM_STATE).ifPresent(c -> updateState(c.getUID(), UnDefType.UNDEF));
+                getThingChannel(CHANNEL_SETPOINT_TEMPERATURE).ifPresent(c -> updateState(c.getUID(), UnDefType.UNDEF));
+                getThingChannel(CHANNEL_DURATION).ifPresent(c -> updateState(c.getUID(), UnDefType.UNDEF));
+            }
+        });
+
+        handlers.put(EVENT_OVEN_CAVITY_TEMPERATURE, event -> {
+            manuallyUpdateCavityTemperature = false;
+            getThingChannel(CHANNEL_OVEN_CURRENT_CAVITY_TEMPERATURE).ifPresent(channel -> updateState(channel.getUID(),
+                    new QuantityType<>(event.getValueAsInt(), mapTemperature(event.getUnit()))));
+        });
+
+        handlers.put(EVENT_SETPOINT_TEMPERATURE,
+                event -> getThingChannel(CHANNEL_SETPOINT_TEMPERATURE)
+                        .ifPresent(channel -> updateState(channel.getUID(),
+                                new QuantityType<>(event.getValueAsInt(), mapTemperature(event.getUnit())))));
+        handlers.put(EVENT_DURATION, event -> getThingChannel(CHANNEL_DURATION).ifPresent(
+                channel -> updateState(channel.getUID(), new QuantityType<>(event.getValueAsInt(), SECOND))));
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (isThingReadyToHandleCommand()) {
+            super.handleCommand(channelUID, command);
+
+            getApiClient().ifPresent(apiClient -> {
+                try {
+                    // turn coffee maker on and standby
+                    if (command instanceof OnOffType && CHANNEL_POWER_STATE.equals(channelUID.getId())) {
+                        apiClient.setPowerState(getThingHaId(),
+                                OnOffType.ON.equals(command) ? STATE_POWER_ON : STATE_POWER_STANDBY);
+                    }
+
+                    @Nullable
+                    String operationState = getOperationState();
+                    if (operationState != null && INACTIVE_STATE.contains(operationState)
+                            && command instanceof QuantityType) {
+                        // set setpoint temperature
+                        if (CHANNEL_SETPOINT_TEMPERATURE.equals(channelUID.getId())) {
+                            @SuppressWarnings("unchecked")
+                            QuantityType<Temperature> quantity = ((QuantityType<Temperature>) command);
+
+                            try {
+                                String value;
+                                String unit;
+
+                                if (quantity.getUnit().equals(SIUnits.CELSIUS)
```

Comment:
This seems to be redundant to the FridgeFreeze handler.

Reply:
moved it to abstract class